lib/IRGen/IRGenSIL.cpp

```diff
@@ -2823,15 +2823,8 @@ void IRGenSILFunction::visitGraphOperationInst(GraphOperationInst *i) {
   }
 
   // Set the device.
-  opDevice = deviceInfo->handleDevicePlacement(opInfo.getOperationName(),
-                                               opDevice, i->getAttributes());
-  assert(!opDevice.empty());
-  if (opDevice != TF_ALL_DEVICES) {
-    auto *setDeviceFn = IGM.getTFE_OpSetDeviceFn();
-    auto device = IGM.getAddrOfGlobalString(opDevice);
-    Builder.CreateCall(setDeviceFn, {op, device, status});
-    checkOk(status);
-  }
+  Builder.CreateCall(IGM.getTFC_OpSetDeviceFromScopeFn(), {op, status});
```

Comment:
are we missing some code, for this change here to work? (otherwise it seems we'll be running everything in CPU -- if our runtime tests passed that way even for GPU build, it'd also bring up the question of how to confirm op execution really happened on GPU. :-) )

in particular, where do we call the new `withDevice`, to set and maintain `deviceScopes` as needed by `getTFC_OpSetDeviceFromScopeFn()`?

Will that require migrating all call-sites (unit tests) away from using the old graph_op insts like `tfc.configureGPU` -- that might be challenging given the graph mode still relies on them. I wonder if we need an intermediate step that allows the old and new device placement mechanisms to co-exist.